test/boost/gossiping_property_file_snitch_test.cc

```diff
@@ -15,78 +15,50 @@
-    cfg.listen_address = my_address;
-    cfg.broadcast_address = my_address;
-    auto snitch_i = std::make_unique<sharded<locator::snitch_ptr>>();
-    auto& snitch = *snitch_i;
+    sharded<snitch_ptr> snitch;
+    {
+        const auto my_address = gms::inet_address("localhost");
+        co_await snitch.start(snitch_config {
+            .name = "org.apache.cassandra.locator.GossipingPropertyFileSnitch",
+            .properties_file_name = (std::filesystem::path(test_files_subdir) / property_fname).string(),
+            .listen_address = my_address,
+            .broadcast_address = my_address
+        });
 
-    return snitch.start(cfg).then([&snitch] {
-        return snitch.invoke_on_all(&locator::snitch_ptr::start);
-    }).then_wrapped([&snitch, exp_result] (auto&& f) -> future<> {
-            try {
-                f.get();
-                if (!exp_result) {
-                    BOOST_ERROR("Failed to catch an error in a malformed "
-                                "configuration file");
-                    return snitch.stop();
-                }
-                auto cpu0_dc = make_lw_shared<sstring>();
-                auto cpu0_rack = make_lw_shared<sstring>();
-                auto res = make_lw_shared<bool>(true);
+        auto start = [&]() {
+            return snitch.invoke_on_all(&snitch_ptr::start);
+        };
+        if (exp_result) {
+            BOOST_CHECK_NO_THROW(co_await start());
+        } else {
+            BOOST_CHECK_THROW(co_await start(), bad_property_file_error);
+        }
+    }
 
-                return snitch.invoke_on(0,
-                        [cpu0_dc, cpu0_rack,
-                         res] (snitch_ptr& inst) {
-                    *cpu0_dc =inst->get_datacenter();
-                    *cpu0_rack = inst->get_rack();
-                }).then([&snitch, cpu0_dc, cpu0_rack, res] {
-                    return snitch.invoke_on_all(
-                            [cpu0_dc, cpu0_rack,
-                             res] (snitch_ptr& inst) {
-                        if (*cpu0_dc != inst->get_datacenter() ||
-                            *cpu0_rack != inst->get_rack()) {
-                            *res = false;
-                        }
-                    }).then([res] {
-                        if (!*res) {
-                            BOOST_ERROR("Data center or Rack do not match on "
-                                        "different shards");
-                        } else {
-                            BOOST_CHECK(true);
-                        }
-                        return make_ready_future<>();
-                    });
-                });
-            } catch (std::exception& e) {
-                BOOST_CHECK(!exp_result);
-                return make_ready_future<>();
-            }
-        }).finally([ snitch_i = std::move(snitch_i) ] () mutable {
-            return snitch_i->stop().finally([snitch_i = std::move(snitch_i)] {});
+    {
```

Comment:
If exp_result is false this block is not supposed to be called, but in new implementation is seems to be called.

Reply:
yes, fixed

Reply:
Silly question. The test got ... broken, but still passed. Does the test make any sense at all?

Reply:
why broken? we just checked that `snitch_base.{_my_dc/_my_rack}` are empty on all shards if we failed to parse the config.

Reply:
:smile:  OK, makes sense. Thanks